apps/astarte_appengine_api/test/support_v2/helpers/database.ex

```diff
@@ -0,0 +1,622 @@
+  """
+
+  ###
+  ### Delete
+  @delete_interface """
+  DELETE FROM :keyspace.interfaces WHERE name = :name
+  """
+
+  @delete_device """
+  DELETE FROM :keyspace.devices WHERE device_id = :device_id
+  """
+
+  @delete_alias """
+  DELETE FROM :keyspace.names
+  WHERE object_name = :object_name
+  AND object_type = 1
+  """
+
+  @delete_group """
+  DELETE FROM :keyspace.grouped_devices WHERE group_name = :group_name
+  """
+
+  @delete_pubkeypem """
+  DELETE FROM :keyspace.kv_store WHERE group = 'auth' AND key = 'jwt_public_key_pem'
+  """
+
+  ###
+  ### Keyspace
+  def create_test_keyspace!(cluster, keyspace) do
+    Xandra.Cluster.execute!(cluster, String.replace(@create_keyspace, ":keyspace", keyspace))
+
+    Xandra.Cluster.execute!(cluster, String.replace(@create_devices_table, ":keyspace", keyspace))
+    Xandra.Cluster.execute!(cluster, String.replace(@create_groups_table, ":keyspace", keyspace))
+    Xandra.Cluster.execute!(cluster, String.replace(@create_names_table, ":keyspace", keyspace))
+    Xandra.Cluster.execute!(cluster, String.replace(@create_kv_store, ":keyspace", keyspace))
+
+    Xandra.Cluster.execute!(
+      cluster,
+      String.replace(@create_endpoints_table, ":keyspace", keyspace)
+    )
+
+    Xandra.Cluster.execute!(
+      cluster,
+      String.replace(@create_individual_properties_table, ":keyspace", keyspace)
+    )
+
+    Xandra.Cluster.execute!(
+      cluster,
+      String.replace(@create_individual_datastreams_table, ":keyspace", keyspace)
+    )
+
+    Xandra.Cluster.execute!(
+      cluster,
+      String.replace(@create_interfaces_table, ":keyspace", keyspace)
+    )
+  end
+
+  def destroy_test_keyspace!(cluster, keyspace) do
+    Xandra.Cluster.execute!(cluster, String.replace(@drop_keyspace, ":keyspace", keyspace))
+  end
+
+  ###
+  ### Insert
+  def insert!(:pubkeypem, cluster, keyspace, pub_key_pem) do
+    query =
+      Xandra.Cluster.prepare!(cluster, String.replace(@insert_pubkeypem, ":keyspace", keyspace))
+
+    Xandra.Cluster.execute!(cluster, query, %{
+      "pem" => pub_key_pem
+    })
+  end
+
+  def insert!(:interface, cluster, keyspace, interfaces) do
+    prepared =
+      Xandra.Cluster.prepare!(cluster, String.replace(@insert_interface, ":keyspace", keyspace))
+
+    batch =
+      Enum.reduce(interfaces, Xandra.Batch.new(), fn interface, acc ->
+        Xandra.Batch.add(acc, prepared, %{
+          "name" => interface.name,
+          "interface_id" => interface.interface_id,
+          "major_version" => interface.major_version,
+          "minor_version" => interface.minor_version,
+          "type" => InterfaceType.to_int(interface.type),
+          "automaton_accepting_states" => :erlang.term_to_binary(:automaton_accepting_states),
+          "automaton_transitions" => :erlang.term_to_binary(:automaton_transitions),
+          "aggregation" => AggregationType.to_int(interface.aggregation),
+          # TODO Mapping to the other table
```

Comment:
Leftover?

Reply:
currently, `select!(:interface)` doesn't join with mapping table. It's need to be done